site/src/pages/AuditPage/AuditFilter.tsx

```diff
@@ -0,0 +1,167 @@
+import { BaseOption } from "components/Filter/options"
+import capitalize from "lodash/capitalize"
+
+const PRESET_FILTERS = [
+  {
+    query: "resource_type:workspace action:create",
+    name: "Created workspaces",
+  },
+  { query: "resource_type:template action:create", name: "Added templates" },
+  { query: "resource_type:user action:delete", name: "Deleted users" },
+  {
+    query: "resource_type:workspace_build action:start build_reason:initiator",
+    name: "Builds started by a user",
+  },
+  {
+    query: "resource_type:api_key action:login",
+    name: "User logins",
+  },
+]
+
+export const AuditFilter = ({
+  filter,
+  error,
+  menus,
+}: {
+  filter: ReturnType<typeof useFilter>
+  error?: unknown
+  menus: {
+    user: UserFilterMenu
+    action: ActionFilterMenu
+    resourceType: ResourceTypeFilterMenu
+  }
+}) => {
+  return (
+    <Filter
+      learnMoreLink="https://coder.com/docs/v2/latest/admin/audit-logs#filtering-logs"
+      presets={PRESET_FILTERS}
+      isLoading={menus.user.isInitializing}
+      filter={filter}
+      error={error}
+      options={
+        <>
+          <ResourceTypeMenu {...menus.resourceType} />
+          <ActionMenu {...menus.action} />
+          <UserMenu {...menus.user} />
+        </>
+      }
+      skeleton={
+        <>
+          <SearchFieldSkeleton />
+          <MenuSkeleton />
+          <MenuSkeleton />
+          <MenuSkeleton />
+        </>
+      }
+    />
+  )
+}
+
+export const useActionFilterMenu = ({
+  value,
+  onChange,
+}: Pick<UseFilterMenuOptions<BaseOption>, "value" | "onChange">) => {
+  const actionOptions: BaseOption[] = AuditActions.map((action) => ({
+    value: action,
+    label: capitalize(action),
+  }))
+  return useFilterMenu({
+    onChange,
+    value,
+    id: "status",
+    getSelectedOption: async () =>
+      actionOptions.find((option) => option.value === value) ?? null,
+    getOptions: async () => actionOptions,
+  })
+}
+
+export type ActionFilterMenu = ReturnType<typeof useActionFilterMenu>
+
+const ActionMenu = (menu: ActionFilterMenu) => {
+  return (
+    <FilterMenu
+      id="action-menu"
+      menu={menu}
+      label={
+        menu.selectedOption ? (
+          <OptionItem option={menu.selectedOption} />
+        ) : (
+          "All actions"
+        )
+      }
+    >
+      {(itemProps) => <OptionItem {...itemProps} />}
+    </FilterMenu>
+  )
+}
+
+export const useResourceTypeFilterMenu = ({
+  value,
+  onChange,
+}: Pick<UseFilterMenuOptions<BaseOption>, "value" | "onChange">) => {
+  const actionOptions: BaseOption[] = ResourceTypes.map((type) => {
+    let label = capitalize(type)
+
+    if (type === "api_key") {
+      label = "API Key"
+    }
+
+    if (type === "git_ssh_key") {
+      label = "Git SSH Key"
+    }
+
+    if (type === "template_version") {
+      label = "Template Version"
+    }
+
+    if (type === "workspace_build") {
+      label = "Workspace Build"
+    }
```

Comment:
This is fine! We could probably shorten with a map if we wanted to, but not a blocker.

Reply:
Yeah, I just went to what Copilot suggested to me 😆 since this is not complex code I think it is ok to let to refactor this if we need to extract it into a function to be reused anywhere else.